### PR TITLE
[BENCH-4616] Update lib constraints to address vuln

### PR DESCRIPTION
**Pull request by dexamundsen**
- io.netty:netty-common: fixed in 4.1.115.Final 
- org.springframework.security:spring-security-*: fixed in 6.3.5

Bumping spring plugin resolves both, and also removes the need for other constraints
indexer and underlay does not include spring plugin, hence need netty explicitly constrained

### CODE REVIEW

File: annotationProcessor/build.gradle [hunk 1]
```diff
@@ -55,16 +55,6 @@ dependencies {
         exclude group: 'io.opentelemetry.instrumentation'
     }
 
-    // explicity declare libs and version to address synk warnings
-    implementation "org.springframework:spring-context:${vSpringWeb}"
```
Comment (dexamundsen) [inline annotationProcessor/build.gradle:59]: Not needed with spring plugin upgrade

File: buildSrc/src/main/groovy/tanagra.java-conventions.gradle [hunk 2]
```diff
@@ -42,14 +42,6 @@ dependencyLocking {
 dependencies {
     // GCP BOM - See https://github.com/GoogleCloudPlatform/cloud-opensource-java/wiki/The-Google-Cloud-Platform-Libraries-BOM
     implementation platform('com.google.cloud:libraries-bom:26.49.0')
-    constraints {
-        // "-jre" for Java 8 or higher
```
Comment (dexamundsen) [inline buildSrc/src/main/groovy/tanagra.java-conventions.gradle:46]: no longer needed with current gcp bom

File: service/build.gradle [hunk 3]
```diff
@@ -58,13 +58,8 @@ configurations.all {
 }
 
 dependencies {
-    // added to address synk warnings
-    constraints {
```
Comment (dexamundsen) [inline service/build.gradle:62]: no longer needed

File: service/build.gradle [hunk 4]
```diff
@@ -109,16 +104,6 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-web'
     implementation 'org.springframework.security:spring-security-core'
 
-    // explicity declare libs and version to address synk warnings
-    implementation "org.springframework:spring-context:${vSpringWeb}"
```
Comment (dexamundsen) [inline service/build.gradle:113]: not needed with spring plugin update

**Review by tjennison-work — APPROVED** [review]

**Pull request merged**